dolphinscheduler-master/src/main/java/org/apache/dolphinscheduler/server/master/engine/task/lifecycle/TaskLifecycleEventType.java

```diff
@@ -38,6 +38,10 @@ public enum TaskLifecycleEventType implements ILifecycleEventType {
      * The task instance is running at the target executor server.
      */
     RUNNING,
+    /**
+     * The task instance's runtime context changed.
+     */
+    RUNTIME_CONTEXT_CHANGED,
```

Comment:
Why need to add this event type? The runtime context should be check in running event or finish event.

Reply:
Import a new event type here can avoid we have multiple event schema for an event type. This event type represent  the task runtime context changed, will not affect the state, if we use running event and finish event, we need to add runtime context info into these events, once runtime context changed, we will need to change these events.

And it's hard to explain why some running events have runtime context but some not. 
So split the runtime context to a single event type can avoid these, this event type doesn't affect the lifecycle of a task.